odoo/addons/base/tests/test_misc.py

```diff
@@ -507,3 +507,71 @@ def test_readonly_dict(self):
             d.update({'baz': 'xyz'})
         with self.assertRaises(TypeError):
             dict.update(d, {'baz': 'xyz'})
+
+class TestFormatLang(TransactionCase):
+    # /!\ The monetary argument does not need to be tested on version <= 17.1 because it has no effect.
+    def test_value_and_digits(self):
+        self.assertEqual(misc.formatLang(self.env, 100.23, digits=1), '100.2')
+        self.assertEqual(misc.formatLang(self.env, 100.23, digits=3), '100.230')
+
+        # If value is an empty string, it should return an empty string (not 0).
+        self.assertEqual(misc.formatLang(self.env, ''), '')
+
+        # If digits is 'None' (default value), it should default to 2.
+        self.assertEqual(misc.formatLang(self.env, 100), '100.00')
+
+        # Default rounding is 'HALF_EVEN'
+        self.assertEqual(misc.formatLang(self.env, 100.205), '100.20')
+        self.assertEqual(misc.formatLang(self.env, 100.215), '100.22')
+
+    def test_grouping(self):
+        self.env["res.lang"].create({
+            "name": "formatLang Test",
+            "code": "fLT",
+            "grouping": "[3,2,-1]",
+            "decimal_point": "!",
+            "thousands_sep": "?",
+        })
+
+        self.env['res.lang']._activate_lang('fLT')
+
+        self.assertEqual(misc.formatLang(self.env['res.lang'].with_context(lang='fLT').env, 1000000, grouping=True), '10?00?000!00')
+        self.assertEqual(misc.formatLang(self.env['res.lang'].with_context(lang='fLT').env, 1000000, grouping=False), '1000000.00')
+
+    def test_decimal_precision(self):
+        self.env['decimal.precision'].create({
+            'name': 'formatLang Test',
+            'digits': 3,  # We want .001 decimals to make sure it is overriden by 'dp'.
+        })
+
+        self.assertEqual(misc.formatLang(self.env, 100, dp='formatLang Test'), '100.000')
```

Comment:
I think we should choose a more "technical style writing for the name.
I didn't understand directly that it was picking the 'decimal.precision' model.
```suggestion
        self.assertEqual(misc.formatLang(self.env, 100, dp='3_numbers_precision'), '100.000')
```